internal/controller/dnsrecord_controller.go

```diff
@@ -659,7 +659,7 @@ func (r *DNSRecordReconciler) applyChanges(ctx context.Context, dnsRecord *v1alp
 	}
 
 	// add related endpoints to the record
-	dnsRecord.Status.ZoneEndpoints = mergeZoneEndpoints(
+	dnsRecord.Status.ZoneEndpoints = common.MergeEndpoints(
```

Comment:
I think you just moved this, with no changes from what i can tell, but I'd rather we looked at a way of getting rid of this entirely. This is combining slices that it thinks should now be in the provider, but really this should just be a list pulled directly from the provider unmodified or something that the plan has calculated and exposed.

Reply:
yes, I did just move it, and I would also like to see it removed eventually. As I mentioned in another comment this was primarily making a test that started failing pass again, I didn't see this as necessary in any of my real world testing.